src/core/stdc/stdio.d

```diff
@@ -1367,13 +1395,17 @@ else version (CRuntime_Microsoft)
   else
   {
     ///
+    pragma(printf)
     int _snprintf(scope char* s, size_t n, scope const char* format, ...);
     ///
+    pragma(printf)
     int  snprintf(scope char* s, size_t n, scope const char* format, ...);
 
     ///
+    pragma(printf)
     int _vsnprintf(scope char* s, size_t n, scope const char* format, va_list arg);
     ///
+    pragma(printf)
     int  vsnprintf(scope char* s, size_t n, scope const char* format, va_list arg);
```

Comment:
@WalterBright : In this case, how will the compiler know to check the 3rd argument ?

Reply:
The last 2 parameters are always the same for the "v" functions.

Reply:
True, but you still rely on a convention. Is it the last `char*` argument, or the argument named format, or some other rule ?

Reply:
The last one is `va_arg`, the second to last is `const(char)*`.